homeassistant/components/assist_satellite/entity.py

```diff
@@ -0,0 +1,152 @@
+        ):
+            self._conversation_id = None
+
+        if self._conversation_id is None:
+            self._conversation_id = ulid.ulid()
+
+        # Update timeout
+        self._conversation_id_time = time.monotonic()
+
+        # Set entity state based on pipeline events
+        self._run_has_tts = False
+
+        await async_pipeline_from_audio_stream(
+            self.hass,
+            context=context,
+            event_callback=self.on_pipeline_event,
+            stt_metadata=stt.SpeechMetadata(
+                language="",  # set in async_pipeline_from_audio_stream
+                format=stt.AudioFormats.WAV,
+                codec=stt.AudioCodecs.PCM,
+                bit_rate=stt.AudioBitRates.BITRATE_16,
+                sample_rate=stt.AudioSampleRates.SAMPLERATE_16000,
+                channel=stt.AudioChannels.CHANNEL_MONO,
+            ),
+            stt_stream=audio_stream,
+            pipeline_id=self._satellite_config.default_pipeline,
+            conversation_id=self._conversation_id,
+            device_id=device_id,
+            tts_audio_output="wav",
+            wake_word_phrase=wake_word_phrase,
+            audio_settings=AudioSettings(
+                silence_seconds=self._satellite_config.finished_speaking_seconds
+            ),
+        )
+
+    def on_pipeline_event(self, event: PipelineEvent) -> None:
+        """Set state based on pipeline stage."""
+        if event.type == PipelineEventType.WAKE_WORD_START:
+            self._set_state(AssistSatelliteState.IDLE)
+        elif event.type == PipelineEventType.STT_START:
+            self._set_state(AssistSatelliteState.LISTENING)
+        elif event.type == PipelineEventType.INTENT_START:
+            self._set_state(AssistSatelliteState.PROCESSING)
+        elif event.type == PipelineEventType.TTS_END:
+            # Wait until tts_response_finished is called to return to idle
+            self._run_has_tts = True
+            self._set_state(AssistSatelliteState.RESPONDING)
+        elif event.type == PipelineEventType.RUN_END:
+            if not self._run_has_tts:
+                self._set_state(AssistSatelliteState.IDLE)
+
+    async def async_get_config(self) -> SatelliteConfig:
+        """Get satellite configuration."""
+        return self._satellite_config
+
+    async def async_set_config(self, config: SatelliteConfig) -> None:
+        """Set satellite configuration."""
+        self._satellite_config = config
+        await self._async_config_updated()
+
+    @abstractmethod
+    async def _async_config_updated(self) -> None:
+        """Inform when the device config is updated.
+
+        Platforms need to make sure that the device has this configuration.
+        """
+
+    @property
+    @abstractmethod
+    def is_microphone_muted(self) -> bool:
+        """Return if the satellite's microphone is muted."""
+
+    @abstractmethod
+    async def async_set_microphone_mute(self, mute: bool) -> None:
```

Comment:
It's not registered as a service , described in services.yaml or used in VoIP. Let's leave it out for now

Reply:
Also shouldn't be abstract as it's guarded behind a supported feature.